qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -252,11 +56,8 @@ def __init__(self):
                 p: self.options.get(p, None) for p in self.__fixed_parameters__
             }
 
-        #: List[CurveData]: Processed experiment data set.
-        self.__processed_data_set = list()
-
-        #: List[int]: Index of physical qubits
-        self._physical_qubits = None
+        #: List[CurveData]: Processed experiment data set. For backward compatibility.
+        self.__processed_data_set = {}
```

Comment:
I would like to be able to remove this at some point so we aren't storing state of whatever the last data analyzed was, but that might be beyond the score of this PR.

Reply:
Yes, this will be removed. I don't think we can remove this now since I expect some users may have custom class using `self._data` method (I encouraged to use this method to get data in old API doc). Instead I just injected deprecation warning.